docs/user/datastores/index_config.rst

```diff
@@ -0,0 +1,171 @@
+            option.one = "foo"
+          }
+        }
+      }
+    }
+
+.. _customizing_index_creation:
+
+Customizing Index Creation
+--------------------------
+
+To speed up ingestion, or because you are only using certain query patterns, you may disable some indices.
+The indices are created when calling ``createSchema``. If nothing is specified, the Z2/Z3 (or XZ2/XZ3
+depending on geometry type) indices and record indices will all be created, as well as any attribute
+indices you have defined.
+
+.. warning::
+
+    Certain queries may be much slower if you disable any indices.
+
+To enable only certain indices, you may set a hint in your simple feature type. The hint key is
+``geomesa.indexes.enabled``, and it should contain a comma-delimited list containing a subset of:
+
+- ``z2`` - corresponds to the Z2 index
+- ``z3`` - corresponds to the Z3 index
+- ``records`` - corresponds to the id/record index (for Accumulo data stores)
+- ``id`` - corresponds to the id/record index (for non-Accumulo data stores)
+- ``attr`` - corresponds to the attribute index
+
+You may set the hint using :ref:`set_sft_options`. If you are using the GeoMesa ``SftBuilder``,
+you may instead call the ``withIndexes`` methods:
+
+.. code-block:: scala
+
+    // scala example
+    import org.locationtech.geomesa.utils.geotools.SftBuilder.SftBuilder
+
+    val sft = new SftBuilder()
+        .stringType("name")
+        .date("dtg")
+        .geometry("geom", default = true)
+        .withIndexes(List("records", "z3", "attr_idx"))
+        .build("mySft")
+
+.. _configuring_z_shards:
+
+Configuring Z-Index Shards
+--------------------------
+
+GeoMesa allows configuration of the number of shards (or splits) into which the Z2/Z3/XZ2/XZ3 indices are
+divided. This parameter may be changed individually for each ``SimpleFeatureType``. If nothing is specified,
+GeoMesa will default to 4 shards. The number of shards must be between 1 and 127.
+
+Shards allow us to pre-split tables, which provides some initial parallelism for reads and writes. As more data is
+written, tables will generally split based on size, thus obviating the need for explicit shards. For small data sets,
+shards are more important as the tables might never split due to size. Setting the number of shards too high can
+reduce performance, as it requires more calculations to be performed per query.
+
+The number of shards is set when calling ``createSchema``. It may be specified through the simple feature type
+user data using the hint ``geomesa.z.splits``. See :ref:`set_sft_options`.
+
+.. code-block:: java
+
+    sft.getUserData().put("geomesa.z.splits", "4");
+
+.. _customizing_z_index:
+
+Configuring Z-Index Time Interval
+---------------------------------
+
+GeoMesa uses a z-curve index for time-based queries. By default, time is split into week-long chunks and indexed
+per-week. If your queries are typically much larger or smaller than one week, you may wish to partition at a
```

Comment:
There should not be a hyphen between "per" and "week" here. (See "General Principle 3" here: http://www.apastyle.org/learn/faqs/when-use-hyphen.aspx)